src/main/kotlin/uk/gov/dwp/dataworks/services/TaskDeploymentService.kt

```diff
@@ -330,6 +331,25 @@ class TaskDeploymentService {
         val azkabanUrl = "https://azkaban.workflow-manager.dataworks.dwp.gov.uk?action=login&cognitoToken=" + containerProperties.cognitoToken
         tabs[50] = ContainerTab("Azkaban", azkabanUrl, false)
 
+        val apEnabledUsers = configurationResolver.getStringConfig(ConfigKey.AP_ENABLED_USERS)
+        val apFrontendId = configurationResolver.getStringConfig(ConfigKey.AP_FRONTEND_ID)
+        val apLambdaArn = configurationResolver.getStringConfig(ConfigKey.AP_LAMBDA_ARN)
+
+        // Conditionally add AP tab to Chrome Headless tabs
+        if (apEnabledUsers.contains(containerProperties.userName)) {
+            val jsonPayload = "{\"frontend_id\": \"$apFrontendId\", \"cognito_username\": \"${containerProperties.userName}\"}"
+            val response = awsCommunicator.invokeLambda(apLambdaArn, jsonPayload)
+            if (response != null) {
+                val apLambdaResponseJsonObject = ObjectMapper().readValue<APLambdaResponseJsonObject>(response)
+                var apAuthorizedUrl = apLambdaResponseJsonObject.authorizedUrl
+                if (apAuthorizedUrl.isNotEmpty()) {
+                    // Replace Authorized URL's DNS inherited from VPCE by custom DNS
+                    apAuthorizedUrl = apAuthorizedUrl.replace("""(?<=/{2}[a-z0-9]{14}.)(.*)(?=/)""".toRegex(), "ap.dataworks.dwp.gov.uk")
+                    tabs[60] = ContainerTab("AP", apAuthorizedUrl, false)
+                }
```

Comment:
Are there no else conditions needed here? Currently there is no handling of empty response or response keys - what is meant to happen, if a user is in `apEnabledUsers` but, the lambda returns nothing?

Reply:
The lambda runs within a try statement and returns null if any exception is raised. A null response would be caught by line 342. Line 345 further verifies that the parsing of the reponse to the APLambdaResponseJsonObject has worked. To answer your question, if a user is in `apEnabledUsers` but the lambda returns nothing, the block would be skipped and no new tab would be added.